doc/nrf/releases/release-notes-changelog.rst

```diff
@@ -312,13 +312,22 @@ Modem libraries
   * Updated:
 
     * The Kconfig option :kconfig:option:`CONFIG_NRF_MODEM_LIB_IPC_PRIO_OVERRIDE` is now deprecated.
+    * The Kconfig option :kconfig:option:`CONFIG_NRF_MODEM_LIB_SYS_INIT` is now deprecated.
```

Comment:
It would be nice to see some reasoning for the change somewhere. Whether it should be in PR description or release notes depends on what the reasoning is.

Reply:
Thanks, I've updated the commit messages and PR description.